core/trino-spi/src/main/java/io/trino/spi/type/SqlTime.java

```diff
@@ -39,7 +39,7 @@ public static SqlTime newInstance(int precision, long picos)
         if (rescale(rescale(picos, 12, precision), precision, 12) != picos) {
             throw new IllegalArgumentException("picos contains data beyond specified precision: " + precision);
         }
-        if (picos < 0 || picos > PICOSECONDS_PER_DAY) {
+        if (picos < 0 || picos >= PICOSECONDS_PER_DAY) {
```

Comment:
Please upcate commit message `fix reading TIME when value overflows to 00:00:00`. 
(or is it ok and just this is how github UI is rendering it?)

Reply:
thanks, fixed